src/Engine/Combat/CombatContext.java

```diff
@@ -46,12 +59,45 @@ public CombatContext(GameMap map,
 
     gameMap = map;
     battleRange = pBattleRange;
+
+    boolean attackerMoved = false;
+    if( null != attacker.path )
+      attackerMoved = attacker.path.getPathLength() > 1;
+    else if( null != attacker.unit && null != attacker.coord )
+      attackerMoved = attacker.unit.x != attacker.coord.xCoord || attacker.unit.y != attacker.coord.yCoord;
+
+    if ( map.isLocationValid(attacker.coord))
+    {
+      attacker.setEnvironment(map.getEnvironment(attacker.coord));
+    }
+    if ( map.isLocationValid(defender.coord))
+    {
+      defender.setEnvironment(map.getEnvironment(defender.coord));
+    }
+
+    if( null == attacker.weapon )
+    {
+      attacker.weapon = attacker.unit.chooseWeapon(defender.model, battleRange, attackerMoved);
+    }
+    if( null == defender.weapon )
+    {
+      defender.weapon = defender.unit.chooseWeapon(attacker.model, battleRange, false);
+    }
+
     // Only attacks at point-blank range can be countered
     if( (1 == battleRange) && (null != defender.weapon) )
     {
       canCounter = true;
     }
 
+    if( attacker.mods.isEmpty() )
+      attacker.mods.addAll(attacker.unit.getModifiers());
+    if( defender.mods.isEmpty() )
+      defender.mods.addAll(defender.unit.getModifiers());
```

Comment:
Should this be an unconditional clear/addAll? It seems like preexisting mods would be unexpected in any case.

Reply:
My general thought here was that the caller *may* (but is not required to) specify the actual modifiers to be used.
The obvious signal to that effect is "are there mods specified already?".
This should be relevant later on for the detailed combat calc UI page.

Reply:
I'm not sure I'm comfortable using implicit signals like that. It might be more clear to add a boolean in the constructor to say whether to pull in CO unit modifiers?
Alternatively: when we go to implement it, there may be a more obvious way to do that job (e.g. just use the normal process of applying modifiers to the CO instead of doing funny implicit business).

Reply:
I take it you are lobbying to unconditionally clear-and-add the modifiers, then?

Reply:
Yeah, probably, until and unless it becomes necessary. Maybe even warn if there are modifiers present already.

Reply:
Okay, one problem: I'm already unconditionally adding the modifiers in `UnitContext`'s constructor - as far as I can tell, this code shouldn't ever actually affect anything.
Thoughts?

Reply:
Sounds like we can just nuke this altogether then. Should we also apply them in the constructor, so we don't have to worry about conditionally/double applying them?

Reply:
Uh, what?
I'm already applying modifiers in the `UnitContext` constructor, so I don't think anywhere else is necessary?

Reply:
You are adding them in the constructor, not applying them. It may not make sense to apply them right away anyway though.

Reply:
Oh. Derp.

That's what I was proposing in some of the other discussions we were having.
The trouble with that IMO is that how and when combat modifiers get applied is currently an implementation detail of the combat engine. Changing that would require e.g. telling a `UnitContext` whether it's the defender or attacker *on construction*, which I don't think is compatible with Counter Break.